project3/coordinator/connected_participants.h

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file The connected Participants.
+ */
+#ifndef PROJECT3_CONNECTED_PARTICIPANTS_H
+#define PROJECT3_CONNECTED_PARTICIPANTS_H
+#include <algorithm>
+#include <mutex>
+#include <unordered_set>
+namespace coordinator::connected_participants {
```

Comment:
I think this is going a little overboard with the namespaces. I would just put everything in the `coordinator` namespace. That will make things more readable.